### DISCUSSION TITLE
Predheat Setup Clarifications

### DISCUSSION THREAD

**Discussion by TCWORLD** [springfall2008/batpred#2947] [General]
I'm trying to verify my predheat set up to tune it to be more accurate. The one value I haven't set up yet is the `heat_loss_watts` parameter.

The description for this is given in the documentation:

> Next, we need to work out the number of watts of heat loss in the house, this can be done by looking at the energy consumed when the heating comes on. Pick a period of heating, ideally from the time the temperature starts increasing for a complete hour of increase, looking at the increase in temperature in degrees, add to that static heat loss which is heat_loss_degrees (internal temp - external temp) 1 hour to get the total degrees accounted for. Now divide that by the external temperature difference again / (internal_temp - external_temp) and multiply the final figure by the energy your system consumed in Watts during that period (can be found either from your sensor or just by looking at your energy bill for the same 1 hour period).
>
>The final figure should be the number of watts your house loses per 1 degree of external temperature difference and be set to heat_loss_watts

However, I'm struggling to make heads or tails of that description. Would you be able to try and clarify this, perhaps with formulae rather than a block of text.

From what I can tell, I need to find a nice section of heating. I've also calculated `heat_loss_degrees: 0.0366` from a 5 hour 'off' period as the house was cooling down.

Take for example this - granted only 40min not a full hour, but there is a continuous output power from the heat pump during that period, and the external temperature is pretty stable.

<img width="895" height="812" alt="image" src="https://github.com/user-attachments/assets/6a1a6ff0-16be-45ca-8f33-9702932581e4" />

That should from the description be all the information needed, but I'm confused.

* The line "add to that static heat loss which is heat_loss_degrees (internal temp - external temp) 1 hour to get the total degrees accounted for". It feels like there's something missing from that sentence.
* "multiply the final figure by the energy your system consumed in Watts during that period" - energy consumed is in kWh not W. Also should that be energy consumed or energy delivered - i.e. output power of the heat pump (6kW) or input power (1.4kW).

----

Trying to read between the lines there, is the following interpretation correct:

1. Average external temperature difference: `delta_t = (19.8 + 18.4) / 2 - 6.4 = 12.7 *C`
2. Temperature increase in degrees: `temp_inc = (19.8 - 18.4) = 1.4 *C`
3. Static heat loss: `static_loss = heat_loss_degrees * delta_t * time_hours = 0.0366 * 12.7 * 40/60 = 0.30988` (I think that ends up being `*C` ???)
4. Energy consumed during period, is integral of power over time, though as that is basically constant can just do a simple multiplication of power and time `energy = 6.2kW * 40/60 = 4.13kWh`
5. Then I interpret the text as `heat_loss_watts = (temp_inc + static_loss) / delta_t * energy`? That comes out as `kWh` units, so presumably then multiply by `40/60` as this is a 40 minute window to get `heat_loss_watts = (temp_inc + static_loss) / delta_t * energy * time = (1.4 + 0.30988) / 12.7 * 4.13 * 40/60 = 556.5 W`.

That seems very high to me.

----

If the figure is meant to be power loss per \*C, then based on the heat-loss survey that I had done when the heat pump was installed (actually three different surveys using different tools and metrics giving similar figures), the total loss comes out as 3.91kW at an outside temperature of -3\*C. At an average design temperature of 20*C, that gives 3911 / 23 = 170 W/\*C which seems more inline with the sample value given.

**Comment by TCWORLD** [comment]
So for now I've just use the heat loss survey value converted to per-degree and preheat seems to be within 10% of the actual value which is good enough for me.
